server/lib/service/dv360.dart

```diff
@@ -27,8 +31,19 @@ class DisplayVideo360Client {
         updateMask: 'entityStatus');
   }
 
-  /// Runs the rule to manipulate DV360 line items.
-  Future<void> run(Rule rule) async {
-    await rule.action.run(this);
+  /// Runs the rule to manipulate DV360 line items and logs the result.
+  Future<void> run(Rule rule, String userId, String ruleId) async {
+    try {
+      await rule.action.run(this);
+    } on ApiRequestError catch (e) {
+      // If there is an API error, return the message returned by the API.
+      return await _firestoreClient.addRun(userId, ruleId,
+          isSuccess: false, message: e.message);
+    } catch (e) {
+      // If there is another kind of exception, do not include the message.
+      return await _firestoreClient.addRun(userId, ruleId, isSuccess: false);
```

Comment:
Why we don't log msg is it is not `ApiRequestError`?

Reply:
Our target users are not necessarily developers, so the messages we log should be user-friendly, actionable and understandable. We can expect this for DV360 API error messages, but probably not for lower level exception messages. Also, there might be security issues if we just directly report the raw exception messages.

Reply:
Clear explanation! Maybe you can add a comment to the catch block?

Reply:
Sure, that sounds good!